### PR TITLE
Upload to modrith from gradle

### PR DESCRIPTION
**Pull request by modmuss50**
Also adds a preRelase global var that will be sent for snapshots, in the past I was just changing curseforge to beta.

### CODE REVIEW

File: build.gradle
Comment (i509VCB) [inline build.gradle:387]: I would hook this up to the publish task in all honesty. Something like `tasks.publish.dependsOn(modrinth)` and in general for curse and github releases

File: build.gradle
Comment (i509VCB) [inline build.gradle:392]: This will be fine for now, I do need to go make minotaur use properties instead of public fields - it shouldn't break if I believe when I do that.

Comment (liach) [reply build.gradle:392]: What's wrong with this using env?

Comment (i509VCB) [reply build.gradle:392]: There isn't an issue with the token in the ENV - but rather it's a nasty bit of the gradle plugin and how it exposes it's api to consumers. Not a blocking issue but I plan to neaten up the plugin upstream.

Not related to FAPI here - but Minotaur stuff inside of the plugin.

Typically you will want to use a getter + setter with an `@Input` annotation in gradle, minotaur uses public mutable fields atm which is a bit of code smell. In groovy the `field = value` is essentially adapted into a `get/set` like how kotlin handles var types.

Comment (liach) [reply build.gradle:392]: if you update the plugin dependency gradle will recompile the buildscript, so this is a non-issue on fabric's end

Comment (i509VCB) [reply build.gradle:392]: Yes - and I believe I can do the upstream changes fine in the future without the buildscript breaking

**Review by liach — APPROVED** [review]
Why not approve this then

**Comment by i509VCB** [comment]
I am mainly waiting for a response on the publish task depending on modrinth publish before I set an approval

**Review by OroArmor — APPROVED** [review]

**Review by YTG1234 — APPROVED** [review]

**Comment by liach** [comment]
> 
> 
> I am mainly waiting for a response on the publish task depending on modrinth publish before I set an approval

I personally disagree on making publish depend on curseforge or modrinth. Those are not maven publication targets.

**Pull request merged**
